### PR TITLE
Add new rule of missing_resource_id and isArmCall option.

### PR DESCRIPTION
**Pull request by raych1**
1. new rule of MISSING_RESOURCE_ID: all the GET/PUT resource call should have 'id' returned in response;
2. add an option isArmCall in liveValidator to differentiate the caller in order to enable different rulesets;
3. fix an issue in response header sourcemap.

### CODE REVIEW

File: lib/liveValidation/operationValidator.ts [hunk 1]
```diff
@@ -128,7 +128,12 @@ export const validateSwaggerLiveResponse = async (
     validateContentType(operation.produces!, headers, false, result);
   }
 
-  const ctx = { isResponse: true, includeErrors, statusCode };
+  const ctx = {
+    isResponse: true,
+    includeErrors,
+    statusCode,
+    isGetPutHttpMethods: operation._method === "get" || operation._method === "put",
```
Comment (PhoenixHe-msft) [inline lib/liveValidation/operationValidator.ts:135]: How about rename to "httpMethod" to enable more future scenario

File: lib/swaggerValidator/ajvSchemaValidator.ts [hunk 2]
```diff
@@ -179,7 +183,7 @@ export const ajvErrorToSchemaValidateIssue = (
     message: errInfo.message,
     params: errInfo.params,
     jsonPathsInPayload: [dataPath],
-    schemaPath: err.schemaPath,
+    schemaPath: errInfo.code === "MISSING_RESOURCE_ID" ? "" : err.schemaPath,
```
Comment (PhoenixHe-msft) [inline lib/swaggerValidator/ajvSchemaValidator.ts:186]: Why do we need to hard code this logic?

File: lib/transform/allOfTransformer.ts [hunk 3]
```diff
@@ -18,6 +18,9 @@ const transformAllOfSchema = (schema: Schema, baseSchemas: Set<Schema>, jsonLoad
     transformAllOfSchema(sch, baseSchemas, jsonLoader);
 
     const { properties, required, additionalProperties: aProperties } = sch;
+    if (sch["x-ms-azure-resource"] === true) {
+      schema["x-ms-azure-resource"] = true;
```
Comment (PhoenixHe-msft) [inline lib/transform/allOfTransformer.ts:22]: What if schema["x-ms-azure-resoutce"] is false? Do we intend to overwrite it?

Comment (PhoenixHe-msft) [reply lib/transform/allOfTransformer.ts:22]: Consider use the variable xmsAzureResource to replace the string

File: lib/swaggerValidator/ajvSchemaValidator.ts [hunk 4]
```diff
@@ -50,6 +50,10 @@ export class AjvSchemaValidator implements SchemaValidator {
       ...options,
     });
     ajvEnableAll(this.ajv, loader);
+
+    if (options && (options as any).isArmCall === true) {
```
Comment (PhoenixHe-msft) [inline lib/swaggerValidator/ajvSchemaValidator.ts:54]: Why options as any?

Comment (PhoenixHe-msft) [reply lib/swaggerValidator/ajvSchemaValidator.ts:54]: Consider add interface SchemaValidatorOptions with isArmCall, then LiveValidatorOptions would inherit SchemaValidatorOptions

File: lib/liveValidation/operationValidator.ts (hunk 1 above)
Comment (raych1) [reply lib/liveValidation/operationValidator.ts]: Refactored.

File: lib/swaggerValidator/ajvSchemaValidator.ts (hunk 2 above)
Comment (raych1) [reply lib/swaggerValidator/ajvSchemaValidator.ts:186]: Doesn't want to set it as '#/x-ms-azureresource' and leave it as ''.

File: lib/transform/allOfTransformer.ts (hunk 3 above)
Comment (raych1) [reply lib/transform/allOfTransformer.ts:22]: Override it only when undefined.

File: lib/swaggerValidator/ajvSchemaValidator.ts (hunk 4 above)
Comment (raych1) [reply lib/swaggerValidator/ajvSchemaValidator.ts:54]: LiveValidatorLoaderOptions cannot transform to ajv.Options.

Comment (PhoenixHe-msft) [reply lib/swaggerValidator/ajvSchemaValidator.ts:54]: You need another option not for ajv.Options

**Review by PhoenixHe-msft — APPROVED** [review]

**Pull request merged**